e2e/validation.go

```diff
@@ -46,9 +51,18 @@ func runLiveVMValidators(ctx context.Context, t *testing.T, vmssName, privateIP,
 		desc := validator.Description
 		command := validator.Command
 		isShellBuiltIn := validator.IsShellBuiltIn
+		isNonHostValidator := validator.IsNonHostNetwork
+
 		t.Logf("running live VM validator on %s: %q", vmssName, desc)
 
-		execResult, err := pollExecOnVM(ctx, t, opts.clusterConfig.Kube, privateIP, podName, sshPrivateKey, command, isShellBuiltIn)
+		var execResult *podExecResult
+		var err error
+		// Non Host Validators - meaning we want to execute checks through a pod which is NOT connected to host's network
+		if isNonHostValidator {
+			execResult, err = execOnPrivilegedPod(ctx, opts.clusterConfig.Kube, "default", nonHostPodName, command)
```

Comment:
the difference between `execOnPod` and `execOnPrivilegedPod` is that we wrap the command in `nsenter`, I don't think we actually need that for the new validator?

Reply:
I think generally we should be using `execOnPod` for pod validators - `execOnPrivilegedPod` should really only be used for ssh'ing into a target VM (e.g. only used by `execOnVM`)

Reply:
Can modify that too, one of the errors i ran into was that commands with `ExecOnPod` were not running in bash, which the `execOnPrivilegedPod` does for you, will modify.